### PR TITLE
Fix NXDL source URLs

### LINKED ISSUE #1149: Wrong URL assigned by utils/nxdl2rst.py for contributed_definitions
[Need] to fix the wrong category source URL issue with your suggestion in https://github.com/nexusformat/definitions/pull/1133#issuecomment-1170371459 before the merge then this PR will not change `build/html` output directory significantly. It means refactor [offered by #1133 will] not fix anything.

_Originally posted by @PeterC-DLS in https://github.com/nexusformat/definitions/issues/1133#issuecomment-1171354612_

### PR DESCRIPTION
**Pull request by PeterC-DLS**
Correct location of contributed definitions as category should not be used. Also sort JSON output to mirror YAML.

- Fixes #1149

### CODE REVIEW

**Comment by prjemian** [comment]
Please select Labels, Project, and Milestone for this PR.

**Comment by prjemian** [comment]
Coordinate with #1151.

File: utils/nxdl2rst.py [hunk 1]
```diff
@@ -671,19 +666,18 @@ def print_rst_from_nxdl(nxdl_file):
...
-    anchor_registry.category = SUBDIR_MAP[subdir]
+    anchor_registry.subdir = subdir
 
     # TODO: check for consistency with root.get('category')
     listing_category = {
         "base": "base class",
         "application": "application definition",
         "contributed": "contributed definition",
-    }[subdir]
+    }[category]
```
Comment (woutdenolf) [inline utils/nxdl2rst.py:680]: This is wrong I think. The `category` will never be `contributed`. So you'll have to use the `subdir` here.

File: utils/nxdl2rst.py [hunk 2]
```diff
@@ -792,13 +786,13 @@ def main():
         print(f"Cannot find {nxdl_file}")
         exit()
 
-    print_rst_from_nxdl(nxdl_file)
+    subdir = os.path.basename(os.getcwd())
+    print_rst_from_nxdl(nxdl_file, subdir)
```
Comment (woutdenolf) [inline utils/nxdl2rst.py:790]: In https://github.com/nexusformat/definitions/pull/1151 (a MR that address the same issue) I derive the subdir from `nxdl_file` inside the `print_rst_from_nxdl` function.

**Review by woutdenolf — CHANGES_REQUESTED** [review]
Same as https://github.com/nexusformat/definitions/pull/1151

Pick whatever you are more comfortable with. This PR does have a bug though.

File: utils/nxdl2rst.py [hunk 3]
```diff
@@ -170,7 +165,7 @@ def _write_html(self, contents):
 
     def _write_json(self, contents):
         with open(self.json_file, "w") as f:
-            json.dump(contents, f, indent=4)
+            json.dump(contents, f, indent=4, sort_keys=True)
```
Comment (woutdenolf) [inline utils/nxdl2rst.py:168]: Not related to the issue.

File: utils/nxdl2rst.py (hunk 1 above)
Comment (woutdenolf) [reply utils/nxdl2rst.py:680]: This is what I did: https://github.com/nexusformat/definitions/pull/1151

```python
listing_category = {
        "base_classes": "base class",
        "applications": "application definition",
        "contributed_definitions": "contributed definition",
    }[subdir]
```

**Pull request closed without merging**